### PR TITLE
Update to use the Swift 5.1 standalone parser.

### PR DESCRIPTION
**Pull request by allevato**
This change also cleans up the generate-pipeline code by refactoring it into a few separate source files, since it was getting a bit unwieldy.

The current version of SwiftSyntax in use is 2019-05-13. There isn't a May 13 trunk snapshot; the May 12 one is compatible ([macOS download link](https://swift.org/builds/development/xcode/swift-DEVELOPMENT-SNAPSHOT-2019-05-12-a/swift-DEVELOPMENT-SNAPSHOT-2019-05-12-a-osx.pkg)).

Once you've installed that snapshot, you'll need to set the `TOOLCHAINS` variable to the bundle ID of that toolchain in your environment before running any `swift` commands on the command line; e.g.,

```shell
TOOLCHAINS=org.swift.50201905121a swift build
TOOLCHAINS=org.swift.50201905121a swift package generate-xcodeproj
TOOLCHAINS=org.swift.50201905121a swift test --parallel  # this works now!
```

### CODE REVIEW

File: Sources/SwiftFormat/FormatPipeline.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
```
Comment (dabelknap) [inline Sources/SwiftFormat/FormatPipeline.swift:5]: It looks like some files have been updated to 2019, but most have been untouched. Did you want to update them all at once or incrementally?

File: Sources/SwiftFormatConfiguration/RuleRegistry.swift [hunk 2]
```diff
@@ -14,44 +14,47 @@
...
-    "NoLeadingUnderscores": true,
-    "OnlyOneTrailingClosureArgument": true,
-    "UseLetInEveryBoundCaseVariable": true,
     "UseSynthesizedInitializer": true,
+    "UseTrailingClosure": true,
+    "UseTripleSlashForDocumentationComments": true,
+    "ValidStringLiterals": true,
     "ValidateDocumentationComments": true,
   ]
 }
```
Comment (dabelknap) [inline Sources/SwiftFormatConfiguration/RuleRegistry.swift:55]: Were new rules added? The file is longer now.

File: Sources/generate-pipeline/main.swift [hunk 3]
```diff
@@ -13,216 +13,24 @@
...
-                   .deletingLastPathComponent()
-let rulesDir =  sourcesDir.appendingPathComponent("SwiftFormatRules")
-let outputFile =  sourcesDir.appendingPathComponent("SwiftFormat")
-                            .appendingPathComponent("PopulatePipeline.swift")
-let ruleRegistryFile = sourcesDir.appendingPathComponent("SwiftFormatConfiguration")
-                                 .appendingPathComponent("RuleRegistry.swift")
-let fm = FileManager.default
-
-// These rules will not be added to the pipeline
-let suppressRules = ["UseEarlyExits", "UseWhereClausesInForLoops"]
```
Comment (dabelknap) [inline Sources/generate-pipeline/main.swift:27]: Are these problem rules still being suppressed somehow?

**Review by dabelknap — COMMENTED** [review]
Since these changes only build and run on a dev release of Swift, do we want to merge this into `format` or keep it in a separate branch until 5.1 is officially out?

File: Sources/SwiftFormat/FormatPipeline.swift (hunk 1 above)
Comment (allevato) [reply Sources/SwiftFormat/FormatPipeline.swift:5]: I was somewhat inconsistent here. I tried to use 2019 in new files and files with major edits, but it looks like I missed this one.

I can do a separate PR to update them all; this one is already big enough.

File: Sources/SwiftFormatConfiguration/RuleRegistry.swift (hunk 2 above)
Comment (allevato) [reply Sources/SwiftFormatConfiguration/RuleRegistry.swift:55]: Ah, I see what happened. The changes to generate-pipeline are causing it to pick up rules that subclass `SyntaxFormatRule` but that don't actually have any methods. I'll fix that.

File: Sources/generate-pipeline/main.swift (hunk 3 above)
Comment (allevato) [reply Sources/generate-pipeline/main.swift:27]: Yes, the check was moved into RuleCollector.swift.

**Review by allevato — COMMENTED** [review]
> Since these changes only build and run on a dev release of Swift, do we want to merge this into format or keep it in a separate branch until 5.1 is officially out?

I'd prefer to go ahead and merge them into format; these changes are deep enough that trying to keep two branches updated seems like a pain. This is also the version of SwiftSyntax that we'll be using internally, so there's no benefit for _us_ to maintain the old version.

Swift 5.1 is likely on the horizon anyway with WWDC coming up, so at least we'll be ahead of development instead of behind.

**Comment by dabelknap** [comment]
The formatter builds on Ubuntu 16.04, but it crashes when you attempt to run it:
```
Fatal error: Unexpectedly found nil while unwrapping an Optional value: file /home/dabelknap/swift/.build/checkouts/swift-syntax/Sources/SwiftSyntax/AtomicCounter.swift, line 42
Current stack trace:
0    libswiftCore.so                    0x00007fdcc8a9e6c0 swift_reportError + 50
1    libswiftCore.so                    0x00007fdcc8b0c7f0 _swift_stdlib_reportFatalErrorInFile + 115
2    libswiftCore.so                    0x00007fdcc8a3553e <unavailable> + 3614014
3    libswiftCore.so                    0x00007fdcc8a356b7 <unavailable> + 3614391
4    libswiftCore.so                    0x00007fdcc881530d <unavailable> + 1385229
5    libswiftCore.so                    0x00007fdcc8a09d48 <unavailable> + 3435848
6    libswiftCore.so                    0x00007fdcc88146d9 <unavailable> + 1382105
7    swift-format                       0x0000559efdae6497 <unavailable> + 2606231
8    swift-format                       0x0000559efdae6359 <unavailable> + 2605913
9    libpthread.so.0                    0x00007fdcc84b4a99 <unavailable> + 60057
10   libswiftCore.so                    0x00007fdcc8abebe0 swift_once + 95
11   swift-format                       0x0000559efdae6347 <unavailable> + 2605895
12   swift-format                       0x0000559efdae6329 <unavailable> + 2605865
13   swift-format                       0x0000559efdb49d7f <unavailable> + 3014015
14   swift-format                       0x0000559efdb49d12 <unavailable> + 3013906
15   swift-format                       0x0000559efdb4b326 <unavailable> + 3019558
16   swift-format                       0x0000559efdb4bc8d <unavailable> + 3021965
17   swift-format                       0x0000559efdb4e9d5 <unavailable> + 3033557
18   swift-format                       0x0000559efdb4fb9d <unavailable> + 3038109
19   swift-format                       0x0000559efd9c01f9 <unavailable> + 1401337
20   swift-format                       0x0000559efe0c6bea <unavailable> + 8768490
21   swift-format                       0x0000559efe0c8d00 <unavailable> + 8776960
22   swift-format                       0x0000559efe0c8760 <unavailable> + 8775520
23   libc.so.6                          0x00007fdcc6254740 __libc_start_main + 240
24   swift-format                       0x0000559efd931769 <unavailable> + 817001
Illegal instruction (core dumped)
```
I don't know if there is anything we can do about it since the crash is in SwiftSyntax.

**Comment by allevato** [comment]
> I don't know if there is anything we can do about it since the crash is in SwiftSyntax.

Yeah, `dlsym` is returning NULL for the C helper function it's trying to load at runtime. I'm not sure why that's happening, but I'm going to go ahead and merge this because I don't _think_ it's anything we're doing.

**Pull request merged**